uint16 encoderData
uint16 speed
uint8 hallState
bool direction